### PR TITLE
Add public ecform data

### PR DESCRIPTION
**Pull request by phieronymus**
Updates https://github.com/univapaycast/univapay-form-checkout/issues/85

### CODE REVIEW

File: src/resources/ECForms.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    failureRedirectUrl?: string;
+
+    /* Charge and token data */
+    appId: string;
+    checkout: CheckoutType;
+    paymentType: PaymentShippingType;
+    tokenType: TransactionTokenType;
+    univapayCustomerId: string | null;
+    capture: boolean;
+    captureAt?: string | null;
```
Comment (laxa88) [inline src/resources/ECForms.ts:50]: this `univapayCustomerId: string | null`, it is `null`-able. Does that mean this is optional?

File: src/resources/ECForms.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    email?: string | null;
+    shippingAddressLine1: string | null;
+    shippingAddressLine2: string | null;
+    shippingAddressCity: string | null;
+    shippingAddressState: string | null;
+    shippingAddressZip: string | null;
+    shippingAddressCountryCode: string | null;
+    buyerName: string | null;
+    buyerNameTransliteration: string | null;
+    buyerDateOfBirth: string | null;
```
Comment (laxa88) [inline src/resources/ECForms.ts:97]: what's the difference between the usage of `xxx?: type` and `xxx: type | null`?

**Review by laxa88 — APPROVED** [review]

File: src/resources/ECForms.ts (hunk 2 above)
Comment (phieronymus) [reply src/resources/ECForms.ts:97]: Here is a list of the different combination 🙂 

```tsx
required: Type; // Type
optional?: Type; // Type | undefined
nullable: Type | null // Type | null
optionalNullable?: Type | null // Type | null | undefind
```

`null` means that the property can be `null` and `undefined` means that the property may be `undefined` or not in the object.

File: src/resources/ECForms.ts (hunk 1 above)
Comment (phieronymus) [reply src/resources/ECForms.ts:50]: `univapayCustomerId: string | null,` means that we expect `univapayCustomerId` to always be a property in the object that can be a string or `null` when it does not exist 🙂 

I may review the parameters in another PR to remove the one we wont use in a first time. Just wanted to keep it simple for this PR as this was moved from the private SDK.

**Review by perrin4869 — APPROVED** [review]

**Comment by perrin4869** [comment]
Looking good to me! maybe cool to reuse this type in the API side to keep them consistent. would need some typescript magic like `Omit` etc I think

**Comment by phieronymus** [comment]
Yes. Let me do some cleanup in the next version to have the type we support only 🙂

**Pull request merged**